os/drivers/input/ist415.c

```diff
@@ -135,6 +147,12 @@ static int ist415_get_touch_data(struct ist415_dev_s *dev, FAR void *buf)
 	}
 	touchvdbg("touch_point %d\n", touch_point + 1);
 
+#ifdef CONFIG_TOUCH_CALLBACK
```

Comment:
In case of TOUCH_POLL we have handle buffer NULL condition inside function [touch read()](https://github.com/Samsung/TizenRT/blob/54895ab2ce883876e8bd454e8f0276e9c6e42a2e/os/drivers/input/touchscreen.c#L192)
but in case of TOUCH_CALLBACK this function is getting directly called from here (https://github.com/Samsung/TizenRT/blob/54895ab2ce883876e8bd454e8f0276e9c6e42a2e/os/drivers/input/ist415.c#L272). So we have to explicitly handle this condition.

Reply:
1. The touch_read does not check whether the buffer is null or not.
2. What you linked is https://github.com/Samsung/TizenRT/blob/e5d7ef9b8629882448210d8615255e70fc591db4/os/drivers/input/ist415.c#L259C13-L259C27. This is for callback, not poll case.

Could you check it again?
And even it is really checking like that, I suggest to do the same thing at the same location.
For the same purpose, if we do different location for each, it's not good to maintain.

Reply:
This is the platform code, it should be easy-maintain and common.

Reply:
@anjana348 I left the same comment on @namanjain7 's PR. Let's concern this point for all.
Same purpose, same location regardless of the config difference as possible.

Reply:
It has been fixed.
Now buffer NULL condition is getting checked at same location for both the cases.